### PR TITLE
unity: enable Unity support for 64-Bit integers

### PR DESCRIPTION
**Pull request by malwro**
…egers

## Description

Enables support for 64-Bit unsigned integers in Unity test framework.

## Motivation and Context

Necessary for unit testing of  _libcache_ library which is currently under development.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by nalajcie — CHANGES_REQUESTED** [review]
also: prefix commit title with just `unity` (component name, not filename)

**Review by damianloew — COMMENTED** [review]
Please complete the PR description including checklists

File: unity/unity_internals.h [hunk 1]
```diff
@@ -128,12 +128,9 @@
...
  *-------------------------------------------------------*/
 
-/* Auto-detect 64 Bit Support */
-#ifndef UNITY_SUPPORT_64
-  #if UNITY_LONG_WIDTH == 64 || UNITY_POINTER_WIDTH == 64
-    #define UNITY_SUPPORT_64
-  #endif
-#endif
+/* 64 Bit Support */
+#define UNITY_SUPPORT_64
```
[the remark this answers is not in the document]
Comment (malwro) [reply unity/unity_internals.h]: Thank you, fixed

File: unity/unity_config.h [hunk 2]
```diff
@@ -3,4 +3,6 @@
 
 #define UNITY_FIXTURE_NO_EXTRAS
 
+#define UNITY_SUPPORT_64
```
Comment (nalajcie) [inline unity/unity_config.h:6]: - please add comment WHY it's needed, eg.
```c
/* force support for 64-bit integer assertions even on 32-bit targets */
```

**Review by nalajcie — CHANGES_REQUESTED** [review]
Please note that the commit message is not entirely true, as You're also adding support for signed integers (You've explicitly written `64-Bit unsigned integers`)

**Review by nalajcie — APPROVED** [review]

**Pull request merged**